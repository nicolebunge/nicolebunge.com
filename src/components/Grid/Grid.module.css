.grid {
  margin-left: auto;
  margin-right: auto;
  max-width: var(--grid-max-width);
  width: 90%;
}
