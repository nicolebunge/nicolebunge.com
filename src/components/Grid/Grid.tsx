import clsx from 'clsx';
import React, { HTMLAttributes, ReactNode } from 'react';
import { grid } from './Grid.module.css';

interface GridProps extends HTMLAttributes<HTMLDivElement> {
  children: ReactNode;
}

function Grid(props: GridProps): JSX.Element {
  const { className, ...otherProps } = props;

  return <div className={clsx(className, grid)} {...otherProps} />;
}

export type { GridProps };
export default Grid;
