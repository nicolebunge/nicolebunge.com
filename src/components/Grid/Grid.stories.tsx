import Grid from '.';

const meta = {
  component: Grid,
};

const Default = {
  args: {
    children: 'Grid',
  },
};

export { Default };
export default meta;
