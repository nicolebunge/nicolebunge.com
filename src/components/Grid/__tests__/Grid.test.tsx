import { composeStories } from '@storybook/testing-react';
import { render } from '@testing-library/react';
import React from 'react';
import * as stories from '../Grid.stories';

const { Default } = composeStories(stories);

describe('Grid component', () => {
  it('renders correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
