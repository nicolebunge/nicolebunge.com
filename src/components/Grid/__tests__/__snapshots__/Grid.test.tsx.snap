// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Grid component > renders correctly 1`] = `
<div
  class="grid"
>
  Grid
</div>
`;
